stream.go

```diff
@@ -210,14 +210,14 @@ func newClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 	}
 	// Provide an opportunity for the first RPC to see the first service config
 	// provided by the resolver.
-	if err := cc.waitForResolvedAddrs(ctx); err != nil {
+	nameResolutionDelayed, err := cc.waitForResolvedAddrs(ctx)
```

Comment:
set nameResolutionDelayed = true by default and if err doesn't happen set it to false. That should be enough

Reply:
In case of name resolution doesn't happen, we are returning from newClientStream method. So, we would need to add nameResolutionDelay to ClientConn attribute itself to verify if the name resolution has already failed before.